shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/ApplicationContext.java

```diff
@@ -172,11 +176,48 @@ public static String getEnvironment() {
         return getProperty("ENV", "local");
     }
 
-    public static Path getRootPath() {
+    private static Path getRootPath() {
         return Paths.get(System.getProperty("user.dir")).getParent();
     }
 
+    private static Path getTempPath() {
+        return Paths.get(System.getProperty("java.io.tmpdir"));
+    }
+
     public static Path getExamplesPath() {
         return getRootPath().resolve("examples");
     }
+
+    public static Path createTempFile(String fileName) throws IOException {
+        Path tempFilePath = getTempPath().resolve(fileName);
+
+        if (!Files.exists(tempFilePath)) {
+            if (tempFilePath.getFileSystem().supportedFileAttributeViews().contains("posix")) {
```

Comment:
This line and line 211 could be changed to call a method that checked this, so that the method can contain the explanation for this.